internal/pkg/service/buffer/storage/local/writer/test/benchmark/benchmark.go

```diff
@@ -0,0 +1,140 @@
+package benchmark
+
+import (
+	"context"
+	"github.com/benbjohnson/clock"
+	"github.com/c2h5oh/datasize"
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/storage"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/storage/compression"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/storage/local"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/storage/local/writer"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/storage/local/writer/disksync"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/storage/staging"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/key"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/model/column"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/utctime"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/testhelper"
+	"github.com/keboola/keboola-as-code/internal/pkg/validator"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"os"
+	"sync"
+	"testing"
+	"time"
+)
+
+// WriterBenchmark is a generic benchmark for writer.SliceWriter.
+type WriterBenchmark struct {
+	B *testing.B
+	// Parallelism is number of parallel write operations.
+	Parallelism int
+	FileType    storage.FileType
+	Columns     column.Columns
+	Allocate    datasize.ByteSize
+	Sync        disksync.Config
+	Compression compression.Config
+	// DataChFactory must return the channel with table rows, the channel must be closed after the n reads.
+	DataChFactory func(ctx context.Context, n int, g *RandomStringGenerator) <-chan []any
+}
+
+func (b *WriterBenchmark) Run(ctx context.Context) {
+	// Init random string generator
+	gen := newRandomStringGenerator()
+
+	// Setup logger
+	logger := log.NewDebugLogger()
+	logger.ConnectTo(testhelper.VerboseStdout())
+
+	// Open volume
+	volume, err := writer.OpenVolume(ctx, logger, clock.New(), b.B.TempDir())
+	require.NoError(b.B, err)
+
+	// Create writer
+	sliceWriter, err := volume.NewWriterFor(b.newSlice(volume))
+	require.NoError(b.B, err)
+
+	// Create data channel
+	dataCh := b.DataChFactory(ctx, b.B.N, gen)
+
+	// Run benchmark
+	b.B.ResetTimer()
+	start := time.Now()
+
+	// Write data in parallel, see Parallelism option.
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		// Start b.Parallelism goroutines
+		for i := 0; i < b.Parallelism; i++ {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				// Read from the channel until the b.B.N rows are processed.
+				for row := range dataCh {
+					assert.NoError(b.B, sliceWriter.WriteRow(row))
+				}
+			}()
+		}
+	}()
```

Comment:
Simulujeme 1000 paralelnych volani `Write`, `Write` caka na syncToDisk ak je to zapnute.
Teda zodpoveda to napr. 1000 spojeniam/requestom z HTTP1, alebo 1000 requestom v HTTP2 (tam sa spojenia zdielaju).